lib/reversetunnel/agentpool.go

```diff
@@ -531,6 +531,23 @@ func (p *AgentPool) transport(ctx context.Context, channel ssh.Channel, requests
 		proxySigner:          p.PROXYSigner,
 		forwardClientAddress: true,
 	}
+
+	// If the AgentPool is being used for Proxy to Proxy communication between two clusters, then
+	// we check if the reverse tunnel server is capable of tracking user connections. This allows
+	// the leaf proxy to track sessions that are initiated via the root cluster. Without providing
+	// the user tracker the leaf cluster metrics will be incorrect and graceful shutdown will not
+	// wait for user sessions to be terminated prior to proceeding with the shutdown operation.
+	if p.IsRemoteCluster {
+		type userConnectionTracker interface {
+			TrackUserConnection() (closed func())
+		}
```

Comment:
I tried adding `TrackUserConnection` to `reversetunnelclient.Server` and it seems to work. Why the additional interface?